tests/osquery/test_api_views.py

```diff
@@ -2401,3 +2418,433 @@ def test_delete_query_permission_denied(self):
+        pack_query2 = self.force_packquery()
+        query_name = pack_query.query.name.upper()
+        new_query = self.force_query(query_name=query_name)
+        data = {
+            "pack": pack_query.pack.pk,
+            "query": new_query.pk,
+            "interval": 120
+        }
+        response = self.put_json_data(reverse("osquery_api:pack_query", args=(pack_query2.pk,)), data)
+        pack_query2.refresh_from_db()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), {
+            "id": pack_query2.pk,
+            "slug": slugify(f"{query_name}-{new_query.pk}"),
+            "pack": pack_query.pack.pk,
+            "query": new_query.pk,
+            "interval": 120,
+            "log_removed_actions": False,
+            "snapshot_mode": False,
+            "shard": None,
+            "can_be_denylisted": True,
+            "created_at": pack_query2.created_at.isoformat(),
+            "updated_at": pack_query2.updated_at.isoformat()
+        })
+        self.assertEqual(pack_query2.slug, slugify(f"{query_name}-{new_query.pk}"))
+        self.assertEqual(pack_query2.query, new_query)
+        self.assertEqual(pack_query2.interval, 120)
+
+    def test_update_packquery_log_removed_actions_snapshot_mode_conflict(self):
+        self.set_permissions("osquery.change_packquery")
+        pack_query = self.force_packquery()
+        data = {
+            "pack": pack_query.pack.pk,
+            "query": pack_query.query.pk,
+            "interval": 120,
+            "log_removed_actions": True,
+            "snapshot_mode": True
+        }
+        response = self.put_json_data(reverse("osquery_api:pack_query", args=(pack_query.pk,)), data)
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json(), {
+            "log_removed_actions": ["'log_removed_actions' and 'snapshot_mode' are mutually exclusive"],
+            "snapshot_mode": ["'log_removed_actions' and 'snapshot_mode' are mutually exclusive"]
+        })
+
+    def test_update_packquery_snapshot_mode_compliance_check_conflict(self):
+        self.set_permissions("osquery.change_packquery")
+        pack_query = self.force_packquery(force_snapshot_mode=True, compliance_check=True)
+        data = {
+            "pack": pack_query.pack.pk,
+            "query": pack_query.query.pk,
+            "interval": 120,
+            "snapshot_mode": False
+        }
+        response = self.put_json_data(reverse("osquery_api:pack_query", args=(pack_query.pk,)), data)
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json(), {
+            "snapshot_mode": ["A compliance check query can only be scheduled in 'snapshot' mode."]
+        })
+
+    def test_update_packquery_fields_invalid(self):
+        self.set_permissions("osquery.change_packquery")
+        pack_query = self.force_packquery()
+        data = {
+            "pack": 5,
```

Comment:
not really bullet proof to pick a random low value for the pack id that doesn't exist. Who knows what can happen.